app/upgrades/mainnet/v1/constants.go

```diff
@@ -30,18 +33,57 @@ var Upgrade = upgrades.Upgrade{
 	},
 }
 
+//go:embed core_contract.wasm
+var wasmCode []byte
+
 func CreateUpgradeHandler(
 	mm upgrades.ModuleManager,
 	configurator module.Configurator,
-	_ *keepers.AppKeepers,
+	keepers *keepers.AppKeepers,
 ) upgradetypes.UpgradeHandler {
-	return func(ctx context.Context, _ upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
-		// Add additional upgrade logic when needed
-
-		/*
-		 * migrations are run in module name alphabetical
-		 * ascending order, except x/auth which is run last
-		 */
-		return mm.RunMigrations(ctx, configurator, fromVM)
+	return func(context context.Context, _ upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		ctx := sdk.UnwrapSDKContext(context)
+
+		// Run module migrations.
+		migrations, err := mm.RunMigrations(ctx, configurator, fromVM)
+		if err != nil {
+			return nil, err
+		}
+
+		// Additional upgrade logic for instantiating the Core Contract.
+		securityGroupAddr := "seda1afk9zr2hn2jsac63h4hm60vl9z3e5u69gndzf7c99cqge3vzwjzs026662"
+		adminAddr, err := sdk.AccAddressFromBech32(securityGroupAddr)
+		if err != nil {
+			return nil, err
+		}
+
+		codeID, _, err := keepers.WasmContractKeeper.Create(ctx, adminAddr, wasmCode, nil)
+		if err != nil {
+			return nil, err
+		}
+		contractAddr, _, err := keepers.WasmContractKeeper.Instantiate(
+			ctx, codeID, adminAddr, adminAddr,
+			[]byte(fmt.Sprintf(`{"token":"aseda", "owner": "%s", "chain_id":"%s"}`, securityGroupAddr, ctx.ChainID())),
```

Comment:
Good reminder to think about the default values we want to set on the core contract for staking config and DR timeouts

Reply:
Will create a checklist in the Notion doc!